src/pybind/mgr/snap_schedule/fs/schedule_client.py

```diff
@@ -33,6 +33,7 @@
 # e.g.: scheduled-2022-04-19-05_39_00_UTC (len = "2022-04-19-05_39_00")
 SNAPSHOT_TS_FORMAT_LEN = 19
 SNAPSHOT_PREFIX = 'scheduled'
+MAX_PATH_MISSING_FAULTS = 10
```

Comment:
hmmm.. so the minimum granularity is hourly snapshots, which implies waiting for 10 hours before automatic deactivation. That's fine. However, if the schedule is days, then we would wait for 10 days before deactivation? Can we normalize this and not be dependent on the schedule unit?

Reply:
The dependency on the schedule unit turned out to be an unintentional side effect.
Maybe we could reduce the attempts to some value.
But how do you propose to **normalize** the time to deactivation ? you mean to say 1 day at the most ?
Then there are **monthly** and **yearly** snapshot schedules supported as well.

Reply:
Thinking about this - looks like its mapped to the respective schedule unit, i.e., hourly - 1hr, daily - 24hrs, monthly - 30days, etc. That seems correct, doesn't it? Which means its not designed to have a grace period. That's fine by default. I think you can add functionality to make that configurable.

Reply:
Would this config item be a system-wide config ? or per fs ?
How can I save this config for reuse after mgr restart or to pass on to other remote mgrs ?

Reply:
Ideally, per path, however, a file system wide config should be good enough. You can do that in separate PR.

Reply:
Added a note to the doc about the significance of the **created** field, which seems to be a source of confusion.